benchmark/benchresult/main.go

```diff
@@ -0,0 +1,135 @@
+It will print the comparison result of intersection benchmarks between two files.
+
+*/
+package main
+
+import (
+	"encoding/gob"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc/benchmark/stats"
+)
+
+func createMap(fileName string, m map[string]stats.BenchResults) {
+	f, err := os.Open(fileName)
+	if err != nil {
+		log.Fatalf("Read file %s error: %s\n", fileName, err)
+	}
+	defer f.Close()
+	var data []stats.BenchResults
+	decoder := gob.NewDecoder(f)
+	if err = decoder.Decode(&data); err != nil {
+		log.Fatalf("Decode file %s error: %s\n", fileName, err)
+	}
+	for _, d := range data {
+		m[d.RunMode+"-"+d.Features.String()] = d
+	}
+}
+
+func intChange(title string, val1, val2 int64) string {
+	percentChange := strconv.FormatFloat(float64((val2-val1)*100/val1), 'f', 2, 64) + "% "
+	return fmt.Sprintf("%10s  %12s  %12s   %8s \n", title, strconv.FormatInt(val1, 10),
+		strconv.FormatInt(val2, 10), percentChange)
+}
+
+func timeChange(title int, val1, val2 time.Duration) string {
+	percentChange := strconv.FormatFloat(float64((val2-val1)*100/val1), 'f', 2, 64) + "% "
+	return fmt.Sprintf("%10s  %12s  %12s   %8s \n", strconv.Itoa(title)+" latency", val1.String(),
+		val2.String(), percentChange)
+}
+
+func compareTwoMap(m1, m2 map[string]stats.BenchResults) {
+	for k2, v2 := range m2 {
+		if v1, ok := m1[k2]; ok {
+			changes := k2 + "\n"
+			changes += fmt.Sprintf("%10s  %12s  %12s   %8s \n", "Title", "Before", "After", "Percentage")
+			changes += intChange("Bytes/op", v1.AllocedBytesPerOp, v2.AllocedBytesPerOp)
+			changes += intChange("Allocs/op", v1.AllocsPerOp, v2.AllocsPerOp)
+			changes += timeChange(v1.Latency[1].Percent, v1.Latency[1].Value, v2.Latency[1].Value)
+			changes += timeChange(v1.Latency[2].Percent, v1.Latency[2].Value, v2.Latency[2].Value)
+			fmt.Printf("%s\n", changes)
+		}
+	}
+}
+
+func compareBenchmark(file1, file2 string) {
+	var BenchValueFile1 map[string]stats.BenchResults
+	var BenchValueFile2 map[string]stats.BenchResults
+	BenchValueFile1 = make(map[string]stats.BenchResults)
+	BenchValueFile2 = make(map[string]stats.BenchResults)
+
+	createMap(file1, BenchValueFile1)
+	createMap(file2, BenchValueFile2)
+
+	compareTwoMap(BenchValueFile1, BenchValueFile2)
+}
+
+func printline(benchName, ltc50, ltc90, allocByte, allocsOp interface{}) {
+	fmt.Printf("%-80v%12v%12v%12v%12v\n", benchName, ltc50, ltc90, allocByte, allocsOp)
+}
+
+func formatBenchmark(fileName string) {
+	f, err := os.Open(fileName)
+	if err != nil {
+		log.Fatalf("Read file %s error: %s\n", fileName, err)
+	}
+	defer f.Close()
+	var data []stats.BenchResults
+	decoder := gob.NewDecoder(f)
+	if err = decoder.Decode(&data); err != nil {
+		log.Fatalf("Decode file %s error: %s\n", fileName, err)
+	}
+	if len(data) == 0 {
+		log.Fatalf("No data in file %s\n", fileName)
+	}
+	printPos := data[0].SharedPosion
+	fmt.Println("\n Shared features: \n" + strings.Repeat("-", 20))
```

Comment:
Do you want the leading space?

Remove the trailing space please.

Reply:
Done. deleted.